nginx/datadog_checks/nginx/nginx.py

```diff
@@ -292,26 +301,18 @@ def _flatten_json(cls, metric_base, val, tags):
                 output.extend(cls._flatten_json(metric_base, val2, tags))
 
         elif isinstance(val, bool):
-            # Turn bools into 0/1 values
-            if val:
-                val = 1
-            else:
-                val = 0
-            output.append((metric_base, val, tags, 'gauge'))
+            output.append((metric_base, int(val), tags, 'gauge'))
 
         elif isinstance(val, (int, float, long)):
             output.append((metric_base, val, tags, 'gauge'))
 
         elif isinstance(val, (text_type, str)):
-            # In the new Plus API, timestamps are now formatted strings, some include microseconds, some don't...
-            try:
-                timestamp = time.mktime(datetime.strptime(val, "%Y-%m-%dT%H:%M:%S.%fZ").timetuple())
-                output.append((metric_base, timestamp, tags, 'gauge'))
-            except ValueError:
+            if val[-1] == "Z":
                 try:
-                    timestamp = time.mktime(datetime.strptime(val, "%Y-%m-%dT%H:%M:%SZ").timetuple())
-                    output.append((metric_base, timestamp, tags, 'gauge'))
+                    timestamp = fromisoformat(val[:-1]).strftime("%s")
```

Comment:
Also, we're losing the ms precision when converting to timestamp anyway, so I would do val[:19] instead and only use the first strptime call.

Reply:
The ms precision was already lost. I've added a test to illustrate it anyway.